### DISCUSSION TITLE
Textadept 12.0

### DISCUSSION THREAD

**Discussion by orbitalquark** [orbitalquark/textadept#435] [General]
Textadept 12.0 is released from https://github.com/orbitalquark/textadept/releases/tag/textadept_12.0

Please see the [migration guide][] for upgrading from Textadept 11 to 12.

This is a major new release with lots of changes compared to 11.4. The [changelog][] contains a comprehensive list of all of the changes and bugfixes, but here are some highlights:

- Added Qt version and made it the default for all platforms (GTK is still supported on Linux).
- New default theme, including support for OS dark mode.
- The new platform minimums are Windows 10 and macOS 11 (Linux does not have one).
- Updated key bindings to be more consistent and deterministic across OSes and platforms.
- All language autocompletion and documentation has been delegated to an external Language Server Protocol (LSP) [module][].
- Support for TextMate- and LSP-style snippets.

[migration guide]: https://orbitalquark.github.io/textadept/manual.html#migrating-from-textadept-11-to-12
[changelog]: https://orbitalquark.github.io/textadept/changelog.html
[module]: https://github.com/orbitalquark/textadept-lsp

**Comment by mhwombat** [comment]
Nix and NixOS users, I submitted a pull request for the new version to the nixpkgs repo. You can track the progress here:

https://github.com/NixOS/nixpkgs/pull/214780

**Reply by mhwombat** [reply]
Textadept 12.0 is now available in "unstable" for Nix and NixOS.
